sbi/neural_nets/embedding_nets/SC_embedding.py

```diff
@@ -0,0 +1,288 @@
+        out_channels: Number of output channels.
+        modes: Number of Fourier modes to multiply,
+            at most floor(N/2) + 1.
+    """
+
+    def __init__(self, in_channels: int, out_channels: int, modes: int):
+        super(SpectralConv1d_SMM, self).__init__()
+
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.modes = modes
+
+        self.scale = 1 / (in_channels * out_channels)
+        self.weights1 = nn.Parameter(
+            self.scale
+            * torch.rand(in_channels, out_channels, self.modes, dtype=torch.cfloat)
+        )
+
+    def compl_mul1d(self, input, weights):
+        """
+        Performs complex multiplication in the Fourier domain.
+
+        Args:
+            input: Input tensor of shape (batch, in_channel, modes).
+            weights: Weight tensor of shape (in_channel, out_channel, modes).
+
+        Returns:
+            torch.Tensor: Output tensor of shape (batch, out_channel, modes).
+        """
+        return torch.einsum("bix,iox->box", input, weights)
+
+    def forward(self, x, transform):
+        """
+        Forward pass of the spectral convolution layer.
+
+        Args:
+            x: Input tensor of shape (batch, n_points, channels).
+            transform: Fourier transform operator with forward and inverse methods.
+
+        Returns:
+            The real part of the transformed output tensor
+            with shape (batch, points, conv_channels).
+        """
+        # Compute Fourier coeffcients up to factor of e^(- something constant)
+        x_ft = transform.forward(x.cfloat(), norm='forward')
+        x_ft = x_ft.permute(0, 2, 1)
+        out_ft = self.compl_mul1d(x_ft, self.weights1)
+        x_ft = out_ft.permute(0, 2, 1)
+
+        # Return to physical space
+        x = transform.inverse(x_ft, norm='backward')
+
+        return x.real
+
+    def last_layer(self, x, transform):
+        """
+        Last convolutional layer returning Fourier coefficients to be used as embedding
+
+        Args:
+            x: Input tensor of shape (batch, points, channels).
+            transform: Fourier transform operator with forward and inverse methods.
+
+        Returns:
+            Transformed output tensor of shape (batch, 2*modes, conv_channels).
+        """
+
+        # Compute Fourier coeffcients up to factor of e^(- something constant)
+        x_ft = transform.forward(x.cfloat(), norm='forward')
+
+        x_ft = x_ft.permute(0, 2, 1)
+        x_ft = self.compl_mul1d(x_ft, self.weights1)  # (batch, conv_channels, modes)
+        x_ft = x_ft.permute(0, 2, 1)  # (batch, modes, conv_channels)
+        x_ft = torch.view_as_real(x_ft)  # (batch, modes, conv_channels, 2)
+        x_ft = x_ft.permute(0, 1, 3, 2)
+        x_ft = x_ft.reshape(x.shape[0], 2 * self.modes, self.out_channels)
+
+        return x_ft
+
+
+class SpectralConvEmbedding(nn.Module):
+    def __init__(
+        self,
+        n_points: int,
+        modes: int,
```

Comment:
can you provide a default value here? Ideally the only thing the user **has** to provide is the dimension